partner_activities_smartbutton/models/res_partner.py

```diff
@@ -0,0 +1,21 @@
+###############################################################################
+# For copyright and license notices, see __manifest__.py file in root directory
+###############################################################################
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    activities_count = fields.Integer(
+        compute='_compute_activities_count',
+        string='Activities Count',
+    )
+
+    def _compute_activities_count(self):
+        model = self.env['mail.activity']
+        for partner in self:
+            partner.activities_count = model.search_count([
+                ('partner_id', '=', partner.id),
```

Comment:
Prueba con esto para tener en cuenta las actividades de los hijos:
```suggestion
                ('partner_id', 'child_of', partner.commercial_partner_id.id),
```